src/drawer/DrawerCloseButton.ts

```diff
@@ -0,0 +1,25 @@
+import { createHook, createComponent } from "reakit-system";
+import { useButton, DialogStateReturn, ButtonProps } from "reakit";
+import { callAllHandlers } from "@chakra-ui/utils";
+import { DRAWER_KEYS } from "./__keys";
+
+export const useDrawerCloseButton = createHook<
+  Pick<DialogStateReturn, "hide">,
+  ButtonProps
```

Comment:
Types are not exported like this 🤔  ?

![image](https://user-images.githubusercontent.com/39694575/92472561-44b2b500-f1f7-11ea-9375-b8d33edec9c4.png)

Reply:
Not needed for any other external use & the types are smaller so i inlined them directly.

Reply:
Okay, not needed as of now.

But only useful, when these components are used as hooks. We can type it when need arises.